packages/jssdk/shuffle/zkShuffle.ts

```diff
@@ -1,395 +1,412 @@
-            preprocessedDeck.Delta[0], preprocessedDeck.Delta[1],
-            preprocessedDeck.Selector,
-            plaintext_output.X0, plaintext_output.X1,
-            plaintext_output.delta0, plaintext_output.delta1,
-            plaintext_output.selector,
-            this.encrypt_wasm, this.encrypt_zkey,
-        );
-    }
+  // Queries the current deck from contract, shuffles & generates ZK proof locally, and updates the deck on contract.
+  private async _shuffle(gameId: number) {
+    const numCards = (await this.smc.getNumCards(gameId)).toNumber();
+    let shuffleFullProof = await this.generate_shuffle_proof(gameId);
+    let solidityProof: SolidityProof = packToSolidityProof(
+      shuffleFullProof.proof
+    );
+    await (
+      await this.smc.playerShuffle(gameId, solidityProof, {
+        config: await this.smc.cardConfig(gameId),
+        X0: shuffleFullProof.publicSignals.slice(
+          3 + numCards * 2,
+          3 + numCards * 3
+        ),
+        X1: shuffleFullProof.publicSignals.slice(
+          3 + numCards * 3,
+          3 + numCards * 4
+        ),
+        selector0: { _data: shuffleFullProof.publicSignals[5 + numCards * 4] },
+        selector1: { _data: shuffleFullProof.publicSignals[6 + numCards * 4] },
+      })
+    ).wait();
+  }
 
-    // Queries the current deck from contract, shuffles & generates ZK proof locally, and updates the deck on contract.
-    private async _shuffle(
-        gameId: number
-    ) {
-        const numCards = (await this.smc.getNumCards(gameId)).toNumber()
-        let shuffleFullProof = await this.generate_shuffle_proof(gameId)
-        let solidityProof: SolidityProof = packToSolidityProof(shuffleFullProof.proof);
-        await (await this.smc.playerShuffle(
-            gameId,
-            solidityProof,
-            {
-                config : await this.smc.cardConfig(gameId) ,
-                X0 : shuffleFullProof.publicSignals.slice(3 + numCards * 2, 3 + numCards * 3),
-                X1 : shuffleFullProof.publicSignals.slice(3 + numCards * 3, 3 + numCards * 4),
-                selector0 : { _data : shuffleFullProof.publicSignals[5 + numCards * 4]},
-                selector1 : { _data : shuffleFullProof.publicSignals[6 + numCards * 4]}
-            }
-        )).wait()
-    }
+  async shuffle(gameId: number): Promise<boolean> {
+    const start = Date.now();
+    await this._shuffle(gameId);
+    console.log(
+      "Player ",
+      await this.getPlayerId(gameId),
+      " Shuffled in ",
+      Date.now() - start,
+      "ms"
+    );
+    return true;
+  }
 
-    async shuffle(
-        gameId: number
-    ): Promise<boolean> {
-        const start = Date.now()
-        await this._shuffle(gameId)
-        console.log("Player ", await this.getPlayerId(gameId), " Shuffled in ", Date.now() - start, "ms")
-        return true
-    }
+  async decrypt(gameId: number, cards: number[]) {
+    const numCards = (await this.smc.getNumCards(gameId)).toNumber();
```

Comment:
should we set `numCards` as a field in the class `ZKShuffle`? `numCards` should be a constant for an instantiation of `ZKShuffle` in my understanding.